src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -144,22 +134,29 @@ export const updatePreviouslyViewedObjectsAndSave = (): ThunkAction<
     ...getActiveGenomePreviouslyViewedObjects(state)
   ];
 
-  const existingIndex = previouslyViewedObjects.findIndex(
-    (previouslyViewedObject) =>
-      previouslyViewedObject.object_id === activeEnsObject.object_id
+  const savedEntitiesWithoutCurrentEntity =
+    previouslyViewedObjects?.filter(
+      (entity) => entity.object_id !== activeEnsObject.object_id
+    ) || [];
+
+  const newObject = {
+    genome_id: activeEnsObject.genome_id,
+    object_id: activeEnsObject.object_id,
+    object_type: activeEnsObject.type,
+    label: activeEnsObject.label,
+    versioned_stable_id:
+      activeEnsObject.type === 'gene'
+        ? activeEnsObject.versioned_stable_id
+        : null
+  };
+
+  const updatedObject = [newObject, ...savedEntitiesWithoutCurrentEntity].slice(
```

Comment:
Why is `Object` in the name of this variable when it actually is an array?

Reply:
Thats true. funny me. This is why we need extra pair of eyes  LOLL